src/odin_fastcs/odin_data.py

```diff
@@ -191,12 +191,39 @@ class FrameProcessorAdapterController(OdinDataAdapterController):
 class FrameProcessorPluginController(OdinAdapterController):
     """SubController for a plugin in a frameProcessor application."""
 
+    async def initialise(self):
+        if any("dataset" in p.path for p in self._parameters):
+
+            def __dataset_parameter(param: OdinParameter):
+                return "dataset" in param.path
+
+            dataset_parameters, self._parameters = partition(
+                self._parameters, __dataset_parameter
+            )
+            if dataset_parameters:
+                dataset_controller = FrameProcessorDatasetController(
+                    self._connection, dataset_parameters, f"{self._api_prefix}"
+                )
+                self.register_sub_controller("DS", dataset_controller)
+                await dataset_controller.initialise()
+
+        return await super().initialise()
+
     def _process_parameters(self):
         for parameter in self._parameters:
             # Remove plugin name included in controller base path
             parameter.set_path(parameter.path[1:])
 
 
+class FrameProcessorDatasetController(OdinAdapterController):
+    def _process_parameters(self):
+        for parameter in self._parameters:
+            short_path = parameter.uri[3:]
+            if short_path[0] == "compressed_size":
+                short_path[0] = "compressed"  # slightly shorter PV names
```

Comment:
I think we should fix this by renaming the dataset to `size` as it is in the API (https://media.dectris.com/SIMPLON_APIReference_v1p8.pdf page 29) rather that hacking it here.

Without this, are the PVs still too long if we v the `BLXXY-EA-EIGER-01` prefix?

Reply:
Okay, have undone that change/made the necessary test changes

Reply:
Looks good. Could you fixup the extra commit into the original so that it is just one commit and then we can merge.